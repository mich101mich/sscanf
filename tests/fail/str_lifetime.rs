fn main() {
    let s;
    {
        let input = String::from("hi");
        s = sscanf::sscanf!(input, "{str}").unwrap();
    }
    println!("{}", s);

    #[derive(sscanf::FromScanf)]
    #[sscanf(format = "{}")]
    struct Wrapper<'a>(&'a str);

    let w;
    {
        let input = String::from("hi");
        w = sscanf::sscanf!(input, "{Wrapper}").unwrap();
    }
    println!("{}", w.0);
}
