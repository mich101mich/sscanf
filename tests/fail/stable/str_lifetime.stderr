error[E0597]: `input` does not live long enough
 --> tests/fail/stable/str_lifetime.rs:5:29
  |
4 |         let input = String::from("hi");
  |             ----- binding `input` declared here
5 |         s = sscanf::sscanf!(input, "{str}").unwrap();
  |                             ^^^^^ borrowed value does not live long enough
6 |     }
  |     - `input` dropped here while still borrowed
7 |     println!("{}", s);
  |                    - borrow later used here

error[E0597]: `input` does not live long enough
  --> tests/fail/stable/str_lifetime.rs:16:29
   |
15 |         let input = String::from("hi");
   |             ----- binding `input` declared here
16 |         w = sscanf::sscanf!(input, "{Wrapper}").unwrap();
   |                             ^^^^^ borrowed value does not live long enough
17 |     }
   |     - `input` dropped here while still borrowed
18 |     println!("{}", w.0);
   |                    --- borrow later used here
